### ISSUE TITLE
No such file or directory: 'ninja'

### ISSUE THREAD

**Issue by jaraco** [google/pytype#1429]
I'd like to run pytype. I tried two different methods of installation. First, [pip-run](https://pypi.org/project/pip-run):

```
pip-run pytype -- -m pytype read.py
...
FileNotFoundError: [Errno 2] No such file or directory: 'ninja'
```

Looking into the error, it seems that pytype expects `ninja` and its scripts to be installed to a location on the executable path.

I tried also with [pipx](https://pypi.org/project/pipx/):

```
pipx install pytype
...
pytype read.py
...
FileNotFoundError: [Errno 2] No such file or directory: 'ninja'
```

By default, pipx only installs scripts from the top-level package into the `bin` directory.

Installing with `pipx --include-deps pytype` does seem to work around the issue (making `ninja` available on the executable PATH).

I'm not the only [user who encountered the error](https://stackoverflow.com/questions/63585642/pytype-fails-on-ninja-dependency).

Upon inspection, I notice that ninja is available to be invoked via runpy:

```
$ pip-run ninja -- -m ninja --help
usage: ninja [options] [targets...]
...
```

That is, if `ninja` is installed into the Python environment, it's possible to invoke it using `python -m ninja`, avoiding the need for the script launcher and specifically the need to locate it (currently by relying on the PATH).

May I suggest, since `pytype` depends on `ninja` to utilize `runpy` to invoke `ninja` and bypass the dependency on the PATH environment?

Basically, it boils down to a subprocess call similar to:

```diff
- subprocess.run(['ninja', ...])
+ subprocess.run([sys.executable, '-m', 'ninja', ...])
```

This approach would break environments that install pytype and ninja into separate Python environments but expose the scripts as executables. If that's an important case to retain, perhaps both techniques could be used, preferring one and falling back to the other if the first fails.

**Comment by rchen152** [comment]
After doing `pip install pytype`, do `pytype read.py` and `python -m pytype read.py` run successfully for you?

We already have some code here to figure out how ninja should be invoked (https://github.com/google/pytype/blob/90d2ba03db8cb36a579c8c0da42acb308f27120f/pytype/tools/analyze_project/pytype_runner.py#L50-L54), so I'm not sure why that's not working.

**Comment by oprypin** [comment]
pytype works for me under pipx. Closing unless there are other confirmations that there is a problem

**Comment by jaraco** [comment]
Yeah, it's working for me now too. Today, the issue doesn't reproduce, even if I use the current versions from back then:

```
 🐚 py -3.10 -m pip-run pytype==2023.5.8 -- -m pytype read.py
Computing dependencies
Analyzing 1 sources with 0 local dependencies
ninja: Entering directory `.pytype'
ninja: error: '/Users/jaraco/read.py', needed by '/Users/jaraco/.pytype/pyi/read.pyi', missing and no known rule to make it
Leaving directory '.pytype'
```

I have no idea why it was failing at the time.

The only thing I can think is that `importable` was `False` at that check, indicating that `find_spec('ninja')` was `False`. There must have been something wrong with my environment.
